protocols/direct-funding_test.go

```diff
@@ -1,14 +1,93 @@
 package protocols
 
 import (
+	"math/big"
 	"testing"
 
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/types"
 )
 
+// TestNew tests the constructor using a TestState fixture
 func TestNew(t *testing.T) {
 	_, err := NewDirectFundingObjectiveState(state.TestState, state.TestState.Participants[0])
 	if err != nil {
 		t.Error(err)
 	}
 }
+
+// Construct various variables for use in TestUpdate
+var s, _ = NewDirectFundingObjectiveState(state.TestState, state.TestState.Participants[0])
+var dummySignature = state.Signature{
+	R: common.Hex2Bytes(`49d8e91bd182fb4d489bb2d76a6735d494d5bea24e4b51dd95c9d219293312d9`),
+	S: common.Hex2Bytes(`22274a3cec23c31e0c073b3c071cf6e0c21260b0d292a10e6a04257a2d8e87fa`),
+	V: byte(1),
+}
+var dummyStateHash = common.Hash{}
+var stateToSign state.State = s.ExpectedStates[0]
+var stateHash, _ = stateToSign.Hash()
+var privateKeyOfParticipant0 = common.Hex2Bytes(`caab404f975b4620747174a75f08d98b4e5a7053b691b41bcfc0d839d48b7634`)
+var correctSignatureByParticipant, _ = stateToSign.Sign(privateKeyOfParticipant0)
+
+func TestUpdate(t *testing.T) {
+	// First, prepare a new objective using the constructor:
+	s, _ := NewDirectFundingObjectiveState(state.TestState, state.TestState.Participants[0])
+
+	// Next, prepare an event with a mismatched channelId
+	e := ObjectiveEvent{
+		ChannelId: types.Destination{},
+	}
+	// Assert that Updating the objective with such an event returns an error
+	// TODO is this the behaviour we want? Below with the signatures, we prefer a log + NOOP (no error)
```

Comment:
Are there concrete practical differences between a noop that returns an error and a noop that logs one? Each puts the handling / investigation duties back to calling code. Returning the error forces calling code to explicitly handle things.

Reply:
I think the option of not returning an error is making the explicit choice that the calling code does *not* need to explicitly handle the error generated deeper in the call stack. So for a malformed signature, some library code is going to return an error that our protocol needs to explicitly handle. But our protocol is able to say "this is not an error that the calling function -- i.e. the engine -- needs to be concerned with". I do not expect the engine to inspect the logs -- they are there only for debugging or analytics purposes. 

Essentially, I'm considering the pattern where we swallow / ignore an error at run time, but still record it for other non runtime needs.